pkg/auth/oauth/external/keycloak/keycloak_test.go

```diff
@@ -0,0 +1,11 @@
+package keycloak
+
+import (
+	"testing"
+
+	"github.com/openshift/origin/pkg/auth/oauth/external"
+)
+
+func TestKeycloak(t *testing.T) {
+	_ = external.Provider(NewProvider(""))
```

Comment:
This is going to try to load the file `""`, which will log an error and return nil. The main thing this is testing is that your provider satisfies the `external.Provider` interface. You could just do _ = external.Provider(provider{}) to test that.

Might be worth having two methods: 
1. `NewProviderFromConfigFile(filename string) (external.Provider, error)` which loads the file and delegates to:
2. `NewProviderFromConfigData(data []byte) (external.Provider, error)` which does the json parse and extraction, and is easy to unit test